src/commands/signzone.rs

```diff
@@ -1255,12 +1256,29 @@ impl SignZone {
             // we skip that, and we skip RRSIGs as they are output only after
             // the RRset that they cover.
             if self.order_rrsigs_after_the_rtype_they_cover {
-                for rrset in owner_rrs
-                    .rrsets()
-                    .filter(|rrset| !matches!(rrset.rtype(), Rtype::SOA | Rtype::RRSIG))
-                {
+                for rrset in owner_rrs.rrsets().filter(|rrset| {
+                    !(matches!(rrset.rtype(), Rtype::SOA | Rtype::RRSIG)
+                        || self.invoked_as_ldns
```

Comment:
This extension to the condition could do with an explanatory comment, and I had to lookup the precedence of || and && to understand how the logic would be evaluated so for me at least some parentheses would help too.

I don't yet know what your change here is doing or why.